plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/gramine/GenerateGramineSGXManifest.kt

```diff
@@ -0,0 +1,96 @@
+package com.r3.conclave.plugin.enclave.gradle.gramine
+
+import com.r3.conclave.common.internal.PluginUtils.GRAMINE_ENCLAVE_JAR
+import com.r3.conclave.common.internal.PluginUtils.PYTHON_FILE
+import com.r3.conclave.plugin.enclave.gradle.ConclaveTask
+import org.gradle.api.GradleException
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.OutputFile
+import org.gradle.process.ExecResult
+import java.io.File
+import javax.inject.Inject
+
+open class GenerateGramineSGXManifest @Inject constructor(objects: ObjectFactory) : ConclaveTask() {
+    companion object {
+        const val GRAMINE_SGX_SIGN_EXECUTABLE = "gramine-sgx-sign"
+        const val GRAMINE_GET_TOKEN_EXECUTABLE = "gramine-sgx-get-token"
+    }
+
+    @get:InputFile
+    val privateKey: RegularFileProperty = objects.fileProperty()
+
+    @get:InputFile
+    val directManifest: RegularFileProperty = objects.fileProperty()
+
+    @get:InputFile
+    val enclaveJar: RegularFileProperty = objects.fileProperty()
+
+    @get:InputFile
+    @Optional
+    val pythonSourcePath: RegularFileProperty = objects.fileProperty()
+
+    @get:OutputFile
+    val sgxManifest: RegularFileProperty = objects.fileProperty()
+
+    @get:OutputFile
+    val sgxToken: RegularFileProperty = objects.fileProperty()
+
+    @get:OutputFile
+    val sgxSig: RegularFileProperty = objects.fileProperty()
+
+    override fun action() {
+        val manifestPath = directManifest.get().asFile.absolutePath
+        val privateKeyPath = privateKey.get().asFile.absolutePath
+        val outputSGXManifestPath = sgxManifest.asFile.get().parentFile.absolutePath
+        val enclaveDestinationJarName = "${outputSGXManifestPath}/$GRAMINE_ENCLAVE_JAR"
+        val enclaveDestinationJarFile = File(enclaveDestinationJarName)
```

Comment:
Does the Gramine tools require that all the input files exist in the same directory? If so then why are you passing in the absolute paths in the command line args? Shouldn't the file names suffice?

And if a common working directory is required, then use the task's [temporary directory](https://docs.gradle.org/current/dsl/org.gradle.api.Task.html#org.gradle.api.Task:temporaryDir), rather than taking one of the input files' directory.

Reply:
Yes, Gramine require files to be in the same directory. I am indirectly using `gramineBuildDirectory` for that, what is wrong with that? I.e. having the files that we build in the build directory isn't somehow normal? It makes very easy to find the files that we have generated.

Reply:
The task doesn't know that, you've added that assumption in this code. Perhaps you need an `@InputDir` instead of separate input files? Or just use the tasks temp, which exists in the project's build directory.

This isn't an major issue, and I certainly don't want to block this PR on this since I want to try this out. However I do think this isn't clean design.

Reply:
So, to be clear. Not all the files actually need to be in the same dir, but only `enclave.jar` and `enclave.py`, that need to be "reachable" on the same path both at build time and runtime, and I haven't found another (easier) way for them to be somewhere apart from `./` (from the point of view of the working directory during the creation of the `manifest.sgx`)

This is because when we run the hash on them they need to be, at build time, in the same path that they will be at runtime.
Theoretically, I could copy them in my `/usr/local/lib` host directory (or similar), hash them and then use the same mount in Gramine, but this would really feel odd.
So I need to copy them on the same working directory where I run `gramine-sgx-sign` so that I can just use the name `enclave.jar` and `enclave.py` (see the `java.manifest.template`)
The other files (manifest, etc) can stay wherever, and we can pass the absolute path for them.

In relation to the temporary dir: is the temporary dir actually "temporary" though?
```
There are no guarantees that the contents of this directory will be kept beyond the execution of the task
```
I really want to keep the files, as they are useful artifacts, especially to understand what is going on.

Reply:
Your reason for doing this is good, I just think we can implement this better. Not an issue for now.